### PR TITLE
[WIP] user login/signup enhancement

### PR DESCRIPTION
**Pull request by anthonyseu**
- [x] move password hash to backend using bcrypt
- [x] fix comments migration typo
- [x] add unique constraint on username

TODO:
- [x] update user seeders, with a default admin user

### CODE REVIEW

**Comment by anthonyseu** [comment]
@jacgrady1 @quiz1991

File: package.json [hunk 1]
```diff
@@ -51,7 +51,8 @@
     "serialize-javascript": "^1.0.0",
     "serve-favicon": "^2.1.6",
     "multer": "1.3.0",
-    "fast-csv": "2.3.0"
+    "fast-csv": "2.3.0",
+    "bcrypt": "v1.0.2"
```
Comment (jacgrady1) [inline package.json]: why prefer bcrpyt over sha

Comment (anthonyseu) [inline package.json]: it is more secure than sha1, sha2

Comment (anthonyseu) [inline package.json]: https://codahale.com/how-to-safely-store-a-password/

**Comment by anthonyseu** [comment]
@jacgrady1 @quiz1991

File: package.json (hunk 1 above)
Comment (jacgrady1) [inline package.json]: 👍

**Comment by jacgrady1** [comment]
👍

**Pull request merged**